wallet/wallet-controller/src/lib.rs

```diff
@@ -621,6 +653,31 @@ impl<T: NodeInterface + Clone + Send + Sync + 'static, W: WalletEvents> Controll
             }
 
             tokio::time::sleep(NORMAL_DELAY).await;
+
+            self.rebroadcast_txs(&mut rebroadcast_txs_timer);
+        }
+    }
+
+    /// Rebroadcast not confirmed transactions
+    fn rebroadcast_txs(&mut self, rebroadcast_txs_timer: &mut Duration) {
+        if get_time() >= *rebroadcast_txs_timer {
+            let _ = self
+                .wallet
+                .get_transactions_to_be_broadcast()
+                .map(|txs| async {
+                    for tx in txs {
+                        let tx_id = tx.transaction().get_id();
+                        let res = self.rpc_client.submit_transaction(tx).await;
+                        if let Err(e) = res {
+                            log::warn!("Rebroadcasting for tx {tx_id} failed: {e}");
+                        }
+                    }
+                })
+                .log_err_pfx("Fetching transactions for rebroadcasting failed:");
+
+            // Reset the timer with a new random interval between 2 and 5 minutes
+            let sleep_interval_sec = make_pseudo_rng().gen_range(120..=300);
```

Comment:
This is a bad idea. Pick a number and sleep it. Don't introduce random behavior unnecessarily.